### PR TITLE
Migrate Prometheus to the new MetricsExporter interface

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: exporter/prometheusexporter/prometheus.go [hunk 1]
```diff
@@ -37,19 +37,20 @@ type prometheusExporter struct {
...
 
-func (pe *prometheusExporter) ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error {
-	merged := make(map[string]*metricspb.Metric)
-	for _, metric := range md.Metrics {
-		merge(merged, metric)
-	}
-	for _, metric := range merged {
-		_ = pe.exporter.ExportMetric(ctx, md.Node, md.Resource, metric)
+func (pe *prometheusExporter) ConsumeMetrics(ctx context.Context, md pdata.Metrics) error {
+	ocmds := pdatautil.MetricsToMetricsData(md)
```
Comment (nilebox) [inline exporter/prometheusexporter/prometheus.go:45]: Out of curiousity: what's the long term plan for this? Having to convert the data as the very first step in the "new" metric exporter is pretty confusing.

File: exporter/prometheusexporter/prometheus.go [hunk 2]
```diff
@@ -37,19 +37,20 @@ type prometheusExporter struct {
...
+	for _, ocmd := range ocmds {
+		merged := make(map[string]*metricspb.Metric)
+		for _, metric := range ocmd.Metrics {
+			merge(merged, metric)
+		}
+		for _, metric := range merged {
+			_ = pe.exporter.ExportMetric(ctx, ocmd.Node, ocmd.Resource, metric)
```
Comment (nilebox) [inline exporter/prometheusexporter/prometheus.go:52]: Unrelated to this PR (the old code is the same), but we shouldn't ignore errors.

File: exporter/prometheusexporter/prometheus.go (hunk 1 above)
Comment (bogdandrutu) [reply exporter/prometheusexporter/prometheus.go:45]: The long term is to change to use the new data directly without oc as intermediate step. Right now focus on changing everything to use the new interfaces so at least we are not everywhere in this transition state and service and config code can be changed to only work with the new interfaces

File: exporter/prometheusexporter/prometheus.go (hunk 2 above)
Comment (bogdandrutu) [reply exporter/prometheusexporter/prometheus.go:52]: You commented before to not do unrelated changes :)

**Pull request merged**